client/gutenberg/extensions/presets/jetpack/utils/refresh-registrations.js

```diff
@@ -0,0 +1,38 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import { getBlockType, registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import getJetpackData from './get-jetpack-data';
+
+/**
+ * Refreshes registration of Gutenberg extensions (blocks and plugins)
+ *
+ * Uses block and plugin availability information obtained from the server to conditionally
+ * register and/or unregister blocks and plugins accordingly
+ *
+ * @returns {void}
+ */
+export default function refreshRegistrations() {
+	const extensions = getJetpackData();
+
+	if ( ! extensions ) {
+		return;
+	}
+
+	forEach( extensions, ( { available }, name ) => {
+		// TODO: Discern between blocks and plugins, use [un]registerPlugin for the latter
```

Comment:
Would checking for `settings.render` vs `settings.save/edit` be enough to recognise plugins from blocks?